javaparser-core/src/main/java/com/github/javaparser/ast/body/AnnotationDeclaration.java

```diff
@@ -135,4 +136,9 @@ public ResolvedAnnotationDeclaration resolve() {
     public Optional<AnnotationDeclaration> toAnnotationDeclaration() {
         return Optional.of(this);
     }
+
+    @Override
+    public FieldDeclaration addField(Type type, String name, Modifier.Keyword... modifiers) {
+        throw new IllegalStateException("Cannot add a field to an annotation declaration."); 
```

Comment:
This seems a good idea

Reply:
@ftomassetti you suggested it years ago, see issue :-)

Reply:
I was so wise when I was younger :)